packages/core/src/rules/oas3/operation-4xx-problem-details-rfc7807.ts

```diff
@@ -0,0 +1,42 @@
+import { Oas3Response, Oas3Schema } from 'core/src/typings/openapi';
+import { Oas3Rule } from '../../visitors';
+import { UserContext } from '../../walk';
+import { validateDefinedAndNonEmpty } from '../utils';
+
+/**
+ * Validation according rfc7807 - https://datatracker.ietf.org/doc/html/rfc7807
+ */
+export const Operation4xxProblemDetailsRfc7807: Oas3Rule = () => {
+  return {
+    ResponsesMap(responses: Record<string, object>, { report, location }: UserContext) {
+      const codes = Object.keys(responses);
+      for (const code of codes) {
+        if (!/4[Xx0-9]{2}/.test(code)) continue;
+        const response = responses[code] as Oas3Response;
+        if (!response.content || !response.content['application/problem+json']) {
```

Comment:
Response here can be a `$ref`. It's better to register on `Response` object and `skip` all non-4xx responses.